diff --git a/libgenlock/Android.mk b/libgenlock/Android.mk
index 95ae4c9..8434936 100644
--- a/libgenlock/Android.mk
+++ b/libgenlock/Android.mk
@@ -8,5 +8,8 @@ LOCAL_C_INCLUDES := hardware/qcom/display/libgralloc
 LOCAL_SRC_FILES := genlock.cpp
 LOCAL_CFLAGS:= -DLOG_TAG=\"libgenlock\"
 LOCAL_MODULE_TAGS := optional
+ifeq ($(BOARD_USES_GENLOCK),true)
+  LOCAL_CFLAGS += -DUSE_GENLOCK
+endif
 LOCAL_MODULE := libgenlock
 include $(BUILD_SHARED_LIBRARY)
diff --git a/libgenlock/genlock.cpp b/libgenlock/genlock.cpp
index 6400994..2bbf30c 100644
--- a/libgenlock/genlock.cpp
+++ b/libgenlock/genlock.cpp
@@ -38,10 +38,6 @@
 
 #define GENLOCK_DEVICE "/dev/genlock"
 
-#ifndef USE_GENLOCK
-#define USE_GENLOCK
-#endif
-
 namespace {
     /* Internal function to map the userspace locks to the kernel lock types */
     int get_kernel_lock_type(genlock_lock_type lockType)
